src/lib.rs

```diff
@@ -808,58 +698,41 @@ pub struct CacheEntry {
     e_tag: String,
 
     /// The number of times the cache entry has been opened.
-    hit_count: int,
+    hit_count: i32,
 
     /// (new in 1.2) A comment provided by the user or the application.
     comment: Option<String>,
 }
 
-impl <S: Encoder<E>, E> Encodable<S, E> for CacheEntry {
-    fn encode(&self, encoder: &mut S) -> Result<(), E> {
-        encoder.emit_struct("CacheEntry", 0, |encoder| {
-            let mut fields: Vec<(&str, Box<Encodable<S, E>>)> = Vec::new();
-            match self.expires {
-                Some(ref expires) => fields.push(("expires", box expires.to_string())),
-                None => ()
-            }
-            fields.push(("lastAccess", box self.last_access.to_string()));
-            fields.push(("eTag", box self.e_tag.to_string()));
-            fields.push(("hitCount", box self.hit_count));
-            match self.comment {
-                Some(ref comment) => fields.push(("comment", box comment.to_string())),
-                None => ()
-            }
-            for (idx, &(name, ref value)) in fields.iter().enumerate() {
-                try!(encoder.emit_struct_field(name, idx, |encoder| (**value).encode(encoder)));
-            }
-            Ok(())
-        })
+impl CacheEntry {
+    pub fn new(
+        expires: Option<String>,
+        last_access: String,
+        e_tag: String,
+        hit_count: i32,
+        comment: Option<String>
+    ) -> CacheEntry {
+        CacheEntry {
+            expires: expires,
+            last_access: last_access,
+            e_tag: e_tag,
+            hit_count: hit_count,
+            comment: comment
+        }
     }
 }
 
 /// A timing value which may be absent or present
 ///
 /// Defaults to -1 in the absent case.
```

Comment:
I think that default only handles the case where the value is missing -- " returns a default value for a field if it’s not present when deserializing". In this case since the value is there (-1) and we want to map it to a different enum value (NotApplicable), I'm thinking we will have to write a custom deserializer. Please correct me if there's a better option!